tools/run-clang-tidy-in-ci.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+set -ex
+
+if [[ ! -d build ]]; then
+  git submodule update --init --recursive
+
+  mkdir build
+  pushd build
+  # We really only need compile_commands.json, so no need to build!
+  time cmake -DBUILD_TORCH=ON ..
+  popd
+
+  # Generate ATen files.
+  time python aten/src/ATen/gen.py \
```

Comment:
Blah... I know you did this in response to a comment I made, but here is *yet another* spot I have to modify if I change how `gen.py` gets called. That's annoying.

Reply:
In a civilized world we would just have a cmake target that does the code generation without the build

Reply:
This should totally work, right? Like, if cmake is in make mode, just `(cd build && make path/to/generated/header/you/want.h)`. Maybe make a synthetic build target for all generated files.